### PR TITLE
Enable tables to take an array of traces and simplify usage

### PR DESCRIPTION
**Pull request by tgsoverly**
To drive consistency, and allow for the selection of data in tables, we need to change tables to allow an array of traces

Todo
- [x] Update table toolbar to include selector

### CODE REVIEW

File: test-projects/demo/dashboards/welcome.visivo.yml [hunk 1]
```diff
@@ -71,9 +71,14 @@ dashboards:
...
-                - header: Widget Type
-                  column: x_data
-                - header: Widget Sales
-                  column: y_data
+                - trace_name: Fibonacci Waterfall Trace
+                  column_defs:
+                    - header: Widget Type
+                      key: columns.x_data
+                    - header: Widget Sales
+                      key: props.y
```
Comment (tgsoverly) [inline test-projects/demo/dashboards/welcome.visivo.yml:82]: I am thinking this way of referencing could make it easier to use and you would have to add column to the trace.  Thoughts?

**Comment by jpjsonic** [comment]
> To drive consistency, and allow for selection of data in tables, we need to change tables to allow an array of traces

We could also allow for data selection in tables by using the material react column filtering. THe other way we could eventually build interactivity into tables is to enable actions on other charts/traces to group and aggregate tables using the material react table functionality.

File: test-projects/demo/dashboards/welcome.visivo.yml (hunk 1 above)
Comment (jpjsonic) [reply test-projects/demo/dashboards/welcome.visivo.yml:82]: Yeah I like this better.

File: test-projects/demo/dashboards/welcome.visivo.yml [hunk 2]
```diff
@@ -71,9 +71,14 @@ dashboards:
           - width: 2
             table:
               name: Fibonacci Waterfall Table
-              trace: ref(Fibonacci Waterfall Trace)
+              traces:
+                - ref(Fibonacci Waterfall Trace)
               columns:
```
Comment (jpjsonic) [inline test-projects/demo/dashboards/welcome.visivo.yml:76]: We should make this optional and pull in all of the trace data if it's not specified.

File: test-projects/demo/dashboards/welcome.visivo.yml [hunk 3]
```diff
@@ -71,9 +71,14 @@ dashboards:
           - width: 2
             table:
               name: Fibonacci Waterfall Table
-              trace: ref(Fibonacci Waterfall Trace)
+              traces:
```
Comment (jpjsonic) [inline test-projects/demo/dashboards/welcome.visivo.yml:74]: Do we want to scope in allowing people to create tables directly from models as well?

File: test-projects/demo/dashboards/welcome.visivo.yml [hunk 4]
```diff
@@ -71,9 +71,14 @@ dashboards:
...
+                - ref(Fibonacci Waterfall Trace)
               columns:
-                - header: Widget Type
-                  column: x_data
-                - header: Widget Sales
-                  column: y_data
+                - trace_name: Fibonacci Waterfall Trace
+                  column_defs:
+                    - header: Widget Type
```
Comment (jpjsonic) [inline test-projects/demo/dashboards/welcome.visivo.yml:79]: We should also make the header optional and just parse the key name pulling out the last value splitting by the dot, removing underscores and capitalizing ie `key.split('.')[-1].replace('_', '').capitalize`

Comment (jpjsonic) [reply test-projects/demo/dashboards/welcome.visivo.yml:79]: So in this example if you didn't specify a header you would get `X Data`

File: viewer/src/components/items/Table.js [hunk 5]
```diff
@@ -1,28 +1,48 @@
 import Loading from "../Loading";
-import React from "react";
+import React, { useState } from "react";
 import { MaterialReactTable } from 'material-react-table';
```
Comment (jpjsonic) [inline viewer/src/components/items/Table.js:3]: Do we want to provide a way for people to enable [aggregation & grouping](https://www.material-react-table.com/docs/guides/aggregation-and-grouping) in this pr while we're working on the tables?

File: test-projects/demo/dashboards/welcome.visivo.yml (hunk 2 above)
Comment (tgsoverly) [reply test-projects/demo/dashboards/welcome.visivo.yml:76]: Perfect, same page.

File: test-projects/demo/dashboards/welcome.visivo.yml (hunk 3 above)
Comment (tgsoverly) [reply test-projects/demo/dashboards/welcome.visivo.yml:74]: I do think that is a good idea, but maybe a follow up?

File: viewer/src/components/items/Table.js (hunk 5 above)
Comment (tgsoverly) [reply viewer/src/components/items/Table.js:3]: I could just turn it on if that is as easy as it needs to be.

File: test-projects/demo/dashboards/welcome.visivo.yml (hunk 2 above)
Comment (tgsoverly) [reply test-projects/demo/dashboards/welcome.visivo.yml:76]: Included.

File: test-projects/demo/dashboards/welcome.visivo.yml (hunk 1 above)
Comment (tgsoverly) [inline test-projects/demo/dashboards/welcome.visivo.yml:82]: Add tests around new names for header and key, and just key.

**Review by jpjsonic — APPROVED** [review]
This is a big one, looks great!

**Review by jpjsonic — APPROVED** [review]
Looked great testing it out!

**Pull request merged**